tests/subsys/fs/multi-fs/src/main.c

```diff
@@ -10,28 +10,27 @@
 #include "test_littlefs.h"
 #include "test_fs_shell.h"
 
-void test_main(void)
+ZTEST(multi_fs, test_multi_fs)
 {
-	ztest_test_suite(multifs_fs_test,
-			 ztest_unit_test(test_clear_flash),
-			 ztest_unit_test(test_fat_mount),
-			 ztest_unit_test(test_littlefs_mount),
-			 ztest_unit_test(test_fat_mkdir),
-			 ztest_unit_test(test_littlefs_mkdir),
-			 ztest_unit_test(test_fat_readdir),
-			 ztest_unit_test(test_littlefs_readdir),
-			 ztest_unit_test(test_fat_rmdir),
-			 ztest_unit_test(test_littlefs_rmdir),
-			 ztest_unit_test(test_fat_open),
-			 ztest_unit_test(test_littlefs_open),
-			 ztest_unit_test(test_fat_write),
-			 ztest_unit_test(test_littlefs_write),
-			 ztest_unit_test(test_fat_read),
-			 ztest_unit_test(test_littlefs_read),
-			 ztest_unit_test(test_fat_close),
-			 ztest_unit_test(test_littlefs_close),
-			 ztest_unit_test(test_fat_unlink),
-			 ztest_unit_test(test_littlefs_unlink),
-			 ztest_unit_test(test_fs_help));
-	ztest_run_test_suite(multifs_fs_test);
+	test_clear_flash();
```

Comment:
Are these all order-dependent? If so we should comment as such. The only reason to put all the tests with in the same ZTEST is if this is a larger end-to-end test that has a bunch of order-dependencies.

Reply:
Ok, I will check it again.

Reply:
I have modified it to minimum order-dependent. And I modify the STACK_SIZE. Because if I didn't modify the STACK_SIZE, it would fail on qemu_x86. So I tried to find the reason, and I try to change it to 2048, it would pass. So I think it should be STACK_SIZE too small that caused it failed on qemu_x86. So I magnify the STACK_SIZE to 2048 to fix this problem. It is same like [https://github.com/zephyrproject-rtos/zephyr/pull/49204](url).

Reply:
Understood.